tools/xmlGenerator/domainGeneratorConnector.cpp

```diff
@@ -0,0 +1,204 @@
+
+private:
+    void addCriteria(std::vector<string> &tokens);
+    void start();
+
+    CParameterMgrFullConnector mConnector;
+    std::unique_ptr<MyLogger> mLogger;
+    std::unique_ptr<CommandHandlerInterface> mCommandHandler;
+};
+
+void XmlGenerator::addCriteria(std::vector<string> &tokens)
+{
+    if (tokens.size() < 3) {
+        throw runtime_error("Not enough arguments to criterion creation request");
+    }
+
+    auto inclusiveness = tokens.front() == "inclusive";
+    tokens.erase(begin(tokens));
+
+    auto name = tokens.front();
+    tokens.erase(begin(tokens));
+
+    auto criterionType = mConnector.createSelectionCriterionType(inclusiveness);
+    if (criterionType == nullptr) {
+        throw runtime_error("Failed to create an " +
+                        string(inclusiveness ? "inclusive" : "exclusive") +
+                        " criterion type");
+    }
+
+    int index = 0;
+    for (const auto &literalValue : tokens) {
+        // inclusive criteria are bitfields
+        int numericalCriterionValue = inclusiveness ? 1 << index : index;
+        string error;
+        bool success = criterionType->addValuePair(numericalCriterionValue, literalValue, error);
+
+        if (not success) {
+            std::ostringstream message;
+            message << "Valuepair '" << numericalCriterionValue << "/" << literalValue
+                    << " rejected for " << name << ". (" << error << ")";
+            throw runtime_error(message.str());
+        }
+        index++;
+    }
+
+    // We don't need to keep a reference to the criterion - no need to store
+    // the returned pointer.
+    if (mConnector.createSelectionCriterion(name, criterionType) == nullptr) {
+        throw runtime_error("Failed to create criterion '" + name + "'");
+    }
+}
+
+void XmlGenerator::parse(std::istream &input)
+{
+    // Read each command from the input
+    // if it is a "createSelectionCriterion" command, apply special
+    // treatment to handle it ourselves instead of passing the command to the
+    // command handler.
+    // Such lines will look like:
+    //     createSelectionCriterion inclusive|exclusive <name> <value> [value, ...]
+    // If it is a "start", it means that the caller has finished creating the
+    // criteria and the Parameter Framework must be started.
+    string line;
+    while (not input.eof()) {
+        std::getline(std::cin, line);
+
+        auto tokens = Tokenizer(line, string(1, '\0')).split();
+        if (tokens.empty()) {
+            continue;
+        }
+        auto command = tokens.front();
+        tokens.erase(begin(tokens)); // drop the command name
+
+        if (command == "createSelectionCriterion") {
+            addCriteria(tokens);
+        } else if (command == "start") {
+            start();
+        } else {
+            string output;
+            if (not mCommandHandler->process(command, tokens, output)) {
+                std::cerr << output << std::endl;
```

Comment:
:+1: The code now continues on error but keep track of how many occurred and report it with the exit code.